lib.nix

```diff
@@ -92,7 +93,14 @@ rec
                   checkout = builtins.fetchGit ({
                     url = v.git;
                   } // lib.optionalAttrs (v ? rev) {
-                    rev = v.rev;
+                    rev = let
+                            query = p: p.name == k && (lib.substring 0 (4 + lib.stringLength v.git) p.source) == "git+${v.git}";
+                            extractRevision = url: lib.last (lib.splitString "#" url);
```

Comment:
nitpick: for future proofing we could grab anything between `#` and `&`, but this is good enough for now!